parl/common/replay_buffer.py

```diff
@@ -0,0 +1,115 @@
+#   Copyright (c) 2018 PaddlePaddle Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import copy
+import random
+from error_handling import LastElementError
+
+
+class Experience(object):
+    def __init__(self, obs, action, reward, episode_end):
+        self.obs = obs
+        self.reward = reward
+        self.action = action
+        self.episode_end = episode_end
+
+
+class Sample(object):
+    def __init__(self, i, n):
+        self.i = i
+        self.n = n
+
+    def __repr__(self):
+        return str(self.__class__) + ": " + str(self.__dict__)
+
+
+class ReplayBuffer(object):
+    def __init__(self, capacity, exp_type=Experience):
+        """
+        Create Replay buffer.
+
+        Args:
+            exp_type(object): Experience class used in the buffer.
+            capacity(int): Max number of experience to store in the buffer. When
+                the buffer overflows the old memories are dropped.
+        """
+        # TODO: check exp_type is a type of Experience
+        self._buffer = []  # a circular queue to store experiences
```

Comment:
Let's make sure that the style is consistent. Do we really want to add "_" before each class member?

Reply:
I wanted to mark it "private", but I think we can get rid of it.

Reply:
done